plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,934 @@
+        ]  # Query all objects
+
+    if state != "query":
+        bgp_peer_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}/bgpPeers/{2}".format(
+            l3out_object.index, node_group_object.index, bgp_peer_objects.index if bgp_peer_objects else "-"
+        )
+
+    ops = []
+
+    if state == "present":
+        peer_prefix_is_empty = check_if_all_elements_are_none(list(peer_prefix.values())) if peer_prefix else True  # peer_prefix is None or empty dict => True
+        import_route_map_is_empty = check_if_all_elements_are_none(list(import_route_map.values())) if import_route_map else True
+        export_route_map_is_empty = check_if_all_elements_are_none(list(export_route_map.values())) if export_route_map else True
+
+        if not peer_prefix_is_empty:
+            peer_prefix_template = initiate_l3out_bgp_peer_mso_templates(mso, "tenant", peer_prefix.get("template"), peer_prefix.get("template_id"))
+            peer_prefix_uuid = mso_template.get_tenant_policy_uuid(peer_prefix_template, peer_prefix.get("name"), "bgpPeerPrefixPolicies")
+
+        if not import_route_map_is_empty:
+            import_route_map_template = initiate_l3out_bgp_peer_mso_templates(
+                mso, "tenant", import_route_map.get("template"), import_route_map.get("template_id")
+            )
+            import_route_map_uuid = mso_template.get_tenant_policy_uuid(import_route_map_template, import_route_map.get("name"), "routeMapPolicies")
+
+        if not export_route_map_is_empty:
+            export_route_map_template = initiate_l3out_bgp_peer_mso_templates(
+                mso, "tenant", export_route_map.get("template"), export_route_map.get("template_id")
+            )
+            export_route_map_uuid = mso_template.get_tenant_policy_uuid(export_route_map_template, export_route_map.get("name"), "routeMapPolicies")
+
+        mso_values = dict(
+            peerAddressV4=ipv4_addr,
+            peerAddressV6=ipv6_addr,
+            adminState=admin_state,
+            authEnabled=True if auth_password else False,
+            allowedSelfASCount=allowed_self_as_count,
+            ebpgMultiHopTTL=ebgp_multi_hop_ttl,
+            siteOfOrigin=site_of_origin,
+            localAsnConfig=local_asn_config,
+            peerPrefixRef=peer_prefix_uuid,
+            importRouteMapRef=import_route_map_uuid,
+            exportRouteMapRef=export_route_map_uuid,
+            password=dict(value=auth_password) if auth_password is not None else None,
+        )
+
+        if remote_asn not in ["", None]:
+            mso_values["peerAsn"] = int(remote_asn)
+
+        if weight not in ["", None]:
+            mso_values["weight"] = int(weight)
+
+        if local_asn not in ["", None]:
+            mso_values["localAsn"] = int(local_asn)
+
+        if not mso.existing:
+            # BGP Controls
+            if bgp_controls:
+                bgp_controls_dict = dict()
+
+                if bgp_controls["allow_self_as"]:
```

Comment:
same value assignment question as above applies to the other dict assignments in your code

Reply:
Handling nested `None` values is not possible with current mso sanitize function.